CONTRIBUTING.md

```diff
@@ -1,5 +1,37 @@
 # Contributing
 
+## Templates update
+
+Every team using the template has shared responsibility and code access to this repository.
+The template will be collectively contributed and versioned.
+
+When a bug fix or a feature change is done on the Go template, a pull request on the Java version 
+(preferred) or an issue will be opened to make sure we keep both versions at the same feature and 
+maturity level.
+
+There is an agreement from each fleet-manager team to update their fleet-manager code on the newest 
+template regularly. Code upates to/from the template onto/from a specific service is expected to be 
+done manually since the two repositories are different. You are free to choose/rely on whichever 
+automation tool you have at your disposal e.g the combination of [git patch](https://git-scm.com/docs/git-format-patch) 
+and [git apply](https://git-scm.com/docs/git-apply) etc. Code fix is done on a roll forward fashion.
+
+### Notifying other of updates
+
+#### From service specific code to template
+
+In case of updates that are coming from a service specific fleet manager to the template, open an issue
+in [ffm project]() stating if it is a bug fix/ enhancement/ version updates etc. To avoid wasting precious 
+time, a PR should only be opened once there is concensus on the adoption of the enhancement. To speed up 
+the decision time, you can start a [zulip thread](https://bf2.zulipchat.com/) or [google group thread](https://groups.google.com/g/factorized-fleet-manager) engaging the interested parties asking if they would like to adopt the proposal. 
+
+#### From template to specific fleet managers
+
+Once a fix / enhancement has landed to the template, an email should be sent to [factorised fleet manager google group](https://groups.google.com/g/factorized-fleet-manager). Likewise, the notification has to be sent to [zulip chat](https://bf2.zulipchat.com/).
+
+>NOTE: Prefer an approach/ a process where the templates and service specific fleet managers are to be updated regulary. 
+This will make the rather "manual" update process less painful to work with by avoiding huge drifts which are likely to 
+cause a painful bing-bang update.
```

Comment:
`big-bang`

Reply:
Good catch! Thanks, to think I went over this part several times ha ha